src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -2,48 +2,2155 @@
-            var source = """
+            var outer = comp.GetMember<NamedTypeSymbol>("Outer");
+            Assert.Equal(Accessibility.Internal, outer.DeclaredAccessibility);
+            Assert.True(((SourceMemberContainerTypeSymbol)outer).IsFile);
+
+            var classC = comp.GetMember<NamedTypeSymbol>("Outer.C");
+            Assert.Equal(Accessibility.Private, classC.DeclaredAccessibility);
+            Assert.False(((SourceMemberContainerTypeSymbol)classC).IsFile);
+        }
+    }
+
+    [Fact]
+    public void Nested_03()
+    {
+        var source = """
+            file class Outer
+            {
                 file class C { }
-                """;
+            }
+            """;
 
-            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular10);
-            comp.VerifyDiagnostics(
-                // (1,12): error CS8652: The feature 'file types' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
-                // file class C { }
-                Diagnostic(ErrorCode.ERR_FeatureInPreview, "C").WithArguments("file types").WithLocation(1, 12));
+        // PROTOTYPE(ft): determine whether an inner file class within a file class is an error or if it's just fine.
+        var comp = CreateCompilation(source, parseOptions: TestOptions.Regular10);
+        comp.VerifyDiagnostics(
+            // (1,12): error CS8652: The feature 'file types' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+            // file class Outer
+            Diagnostic(ErrorCode.ERR_FeatureInPreview, "Outer").WithArguments("file types").WithLocation(1, 12),
+            // (3,16): error CS8652: The feature 'file types' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+            //     file class C { }
+            Diagnostic(ErrorCode.ERR_FeatureInPreview, "C").WithArguments("file types").WithLocation(3, 16));
+
+        comp = CreateCompilation(source);
+        comp.VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void Nested_04()
+    {
+        var source = """
+            file class Outer
+            {
+                public class C { }
+            }
+
+            class D
+            {
+                void M(Outer.C c) { } // 1
+            }
+            """;
+
+        var comp = CreateCompilation(source);
+        comp.VerifyDiagnostics(
+            // (8,10): error CS9300: File type 'Outer.C' cannot be used in a member signature in non-file type 'D'.
+            //     void M(Outer.C c) { } // 1
+            Diagnostic(ErrorCode.ERR_FileTypeDisallowedInSignature, "M").WithArguments("Outer.C", "D").WithLocation(8, 10));
+    }
+
+    [Fact]
+    public void Nested_05()
+    {
+        var source = """
+            file class Outer
+            {
+                public class C
+                {
+                    void M(Outer outer) { } // ok
```

Comment:
nit: consider adding `void M2(C c) { } // ok` as well in this test (unless covered elsewhere)